torchao/quantization/quantize_/workflows/intx/intx_unpacked_tensor.py

```diff
@@ -0,0 +1,268 @@
+        block_size: the block size for quantization, representing the granularity, for example groupwise quantization will have block_size (1, group_size)
+    """
+
+    tensor_data_names = ["int_data", "scale", "zero_point"]
+    tensor_attribute_names = ["bit_width", "block_size"]
+
+    def __new__(cls, int_data, scale, zero_point, bit_width, block_size=None):
+        kwargs = {}
+        kwargs["device"] = int_data.device
+        kwargs["dtype"] = scale.dtype
+        kwargs["requires_grad"] = False
+        shape = int_data.shape
+        return torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)  # type: ignore[attr-defined]
+
+    def __init__(
+        self,
+        int_data,
+        scale,
+        zero_point,
+        bit_width,
+        block_size: Optional[Tuple[int]] = None,
+    ):
+        # Check plain data and infer block_size from shapes
+        if block_size is None:
+            assert scale.ndim == int_data.ndim
+            assert zero_point.ndim == int_data.ndim
+            block_size = []
+            for i in range(int_data.ndim):
+                assert scale.shape[i] == zero_point.shape[i]
+                n_blocks = scale.shape[i]
+                assert int_data.shape[i] % n_blocks == 0
+                block_size.append(int_data.shape[i] // n_blocks)
+            block_size = tuple(block_size)
+        else:
+            assert len(block_size) == int_data.ndim
+            n_blocks = []
+            for i in range(len(block_size)):
+                assert int_data.shape[i] % block_size[i] == 0
+                n_blocks.append(int_data.shape[i] // block_size[i])
+            scale = scale.reshape(*n_blocks)
+            zero_point = zero_point.reshape(*n_blocks)
+
+        assert block_size is not None
+        assert isinstance(block_size, tuple)
+        assert bit_width >= 1 and bit_width <= 8
+
+        self.int_data = int_data
+        self.scale = scale
+        self.zero_point = zero_point
+
+        self.bit_width = bit_width
+        self.block_size = block_size
+
+    def __repr__(self):
+        repr_fields = (
+            self.tensor_data_names
+            + self.tensor_attribute_names
+            + ["shape", "device", "dtype", "require_grad"]
+        )
+        inner_repr = [f"{attr}={getattr(self, attr)}" for attr in repr_fields]
+        inner_repr = ", ".join(inner_repr)
+        return f"{self.__class__.__name__}({inner_repr}))"
+
+    def _quantization_type(self):
+        return f"bit_width={self.bit_width}, block_size={self.block_size}, shape={self.shape}, dtype={self.dtype}, device={self.device}"
+
+    def _has_float_zero_point(self) -> bool:
+        return self.zero_point.dtype in _FLOAT_TYPES
+
+    def to(self, *args, **kwargs):
+        kwargs = self._get_to_kwargs(*args, **kwargs)
+        device = kwargs.pop("device")
+        dtype = kwargs.pop("dtype")
+        assert dtype in _FLOAT_TYPES
+        return self.__class__(
```

Comment:
nit: `self.__class__` --> `IntxUnpackedTensor` to reduce runtime check and align with other code